src/OpenTelemetry.Exporter.Geneva/TLDExporter/TldLogExporter.cs

```diff
@@ -411,22 +411,26 @@ internal void SerializeLogRecord(LogRecord logRecord)
         partCFieldsCountFromState = dataForScopes.PartCFieldsCountFromState;
 
         int partCFieldsCount = partCFieldsCountFromState + hasEnvProperties; // We at least have these many fields in Part C
-        var partCFieldsCountPatch = eb.AddStruct("PartC", (byte)partCFieldsCount);
 
-        for (int i = 0; i < partCFieldsCountFromState; i++)
+        if (partCFieldsCount > 0)
         {
-            Serialize(eb, kvpArrayForPartCFields[i].Key, kvpArrayForPartCFields[i].Value);
-        }
+            var partCFieldsCountPatch = eb.AddStruct("PartC", (byte)partCFieldsCount);
 
-        if (hasEnvProperties == 1)
-        {
-            // Get all "other" fields and collapse them into single field
-            // named "env_properties".
-            var serializedEnvPropertiesStringAsBytes = JsonSerializer.SerializeKeyValuePairsListAsBytes(envPropertiesList, out var count);
-            eb.AddCountedAnsiString("env_properties", serializedEnvPropertiesStringAsBytes, 0, count);
-        }
+            for (int i = 0; i < partCFieldsCountFromState; i++)
+            {
+                Serialize(eb, kvpArrayForPartCFields[i].Key, kvpArrayForPartCFields[i].Value);
+            }
 
-        eb.SetStructFieldCount(partCFieldsCountPatch, (byte)partCFieldsCount);
+            if (hasEnvProperties == 1)
```

Comment:
nit: `hasEnvProperties == 1` seems a bit odd. Should it be a `bool` or maybe renamed `envPropertyCount`?

Reply:
That was done just to avoid a cast operation. Probably doesn't save much. 😄

https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/2a9ade7e9e12489ad9ecdb980d5e725932b759ea/src/OpenTelemetry.Exporter.Geneva/TLDExporter/TldTraceExporter.cs#L334